pytition/petition/management/commands/medias_orphan.py

```diff
@@ -0,0 +1,22 @@
+from pathlib import Path
+from django.core.management.base import BaseCommand
+from django.conf import settings
+
+from petition.models import Petition
+
+
+
+class Command(BaseCommand):
+    """find orphans medias in mediaroot
+    """
+    def handle(self, *args, **options):
+        mediaroot = Path(settings.MEDIA_ROOT)
+        all_petition = Petition.objects.all()
+        last_petition = all_petition[len(all_petition)-1]
```

Comment:
Here you could use:
```
last_petition = all_petition[-1]
```

Reply:
The queryset all_petition doesn't accept negative index and raise an exception, It's why i do len(all_petition)-1.

[https://docs.djangoproject.com/fr/2.2/_modules/django/db/models/query/](https://docs.djangoproject.com/fr/2.2/_modules/django/db/models/query/)

```
...
assert ((not isinstance(k, slice) and (k >= 0)) or
                (isinstance(k, slice) and (k.start is None or k.start >= 0) and
                 (k.stop is None or k.stop >= 0))), \
            "Negative indexing is not supported."
...
```

Reply:
Ah you're right, I didn't know negative indexing of QuerySet slices was not possible.
I can't test right now but what about:

```
all_petition = Petition.objects.all()
last_petition = all_petition.last()
```

Reply:
all_petition[len(all_petition)-1] and all_petition.last() are two different objects because the last() method order the queryset by primary key:

[https://docs.djangoproject.com/fr/2.2/_modules/django/db/models/query/](https://docs.djangoproject.com/fr/2.2/_modules/django/db/models/query/)

```
 def last(self):
        """Return the last object of a query or None if no match is found."""
        for obj in (self.reverse() if self.ordered else self.order_by('-pk'))[:1]:
            return obj
```

The use of last() method involve ordering the queryset before the loop:

`all_petition = Petition.objects.all().order_by('-pk')`

I don't know what is better : query the database to order or use the len function ?

Tell me what you prefer.

Reply:
Hi @bbmt-bbmt sorry for the delay I was out there without internet access for some weeks!
I think something like that would work:

```python
all_petition = list(Petition.objects.all())
last_petition = all_petition[-1]
```
Also, I think neither (yours or mine) code work when there is no petition at all.
So I guess we need to do something like

```python
all_petition = list(Petition.objects.all())
if all_petition:
    last_petition = all_petition[-1]
```

and then later on:

```python
        for media in mediaroot.glob('**/*.*'):
            media = str(media)
            if all_petition:
                for petition in all_petition:
                    if media in petition.text or media in petition.twitter_image:
                        break
                    if petition == last_petition:
                        print(media)
            else:
                print(media)
```

Reply:
We can't easyly convert a queryset to a list.
when you do `list(Petition.objects.all())` the list contains only the titles of the petition, not the petitiion class itself with the content

Reply:
To me the list contains Petition objects:

```
(.pytition) fallen@fallen-ThinkPad-X260:~/dev/Pytition$ DJANGO_SETTINGS_MODULE=pytition.settings.maconfig python3 ./pytition/manage.py gen_pet --user fallen --number 10
(.pytition) fallen@fallen-ThinkPad-X260:~/dev/Pytition$ DJANGO_SETTINGS_MODULE=pytition.settings.maconfig python3 ./pytition/manage.py shell
Python 3.8.2 (default, Jul 16 2020, 14:00:26) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> from petition.models import Petition
>>> petitions = Petition.objects.all()
>>> pytition_list = list(petitions)
>>> pytition_list
[Petition 0, Petition 1, Petition 2, Petition 3, Petition 4, Petition 5, Petition 6, Petition 7, Petition 8, Petition 9]
>>> pytition_list[0]
Petition 0
>>> pytition_list[0].title
'Petition 0'
>>> pytition_list[0].text
'content'
>>> pytition_list[0].owner
fallen
>>> type(pytition_list[0])
<class 'petition.models.Petition'>

```

Reply:
Ok for me, I made a mistake. I have just made the correction

Reply:
Thanks a lot!
And sorry for the painful code review!